File: fe2-android/app/src/test/java/com/github/dedis/popstellar/utility/scheduler/TestSchedulerProvider.java
package com.github.dedis.popstellar.utility.scheduler;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.TestScheduler;

public class TestSchedulerProvider implements SchedulerProvider {

  private final TestScheduler testScheduler;

  public TestSchedulerProvider() {
    testScheduler = new TestScheduler();
  }

  @Override
  public Scheduler io() {
    return testScheduler;
  }

  @Override
  public Scheduler computation() {
    return testScheduler;
  }

  @Override
  public Scheduler newThread() {
    return testScheduler;
  }

  @Override
  public Scheduler mainThread() {
    return testScheduler;
  }

  public TestScheduler getTestScheduler() {
    return testScheduler;
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/utility/scheduler/TestSchedulerProvider.kt
package com.github.dedis.popstellar.utility.scheduler

import io.reactivex.Scheduler
import io.reactivex.schedulers.TestScheduler

class TestSchedulerProvider : SchedulerProvider {
  @JvmField val testScheduler: TestScheduler = TestScheduler()

  override fun io(): Scheduler {
    return testScheduler
  }

  override fun computation(): Scheduler {
    return testScheduler
  }

  override fun newThread(): Scheduler {
    return testScheduler
  }

  override fun mainThread(): Scheduler {
    return testScheduler
  }
}
